### ISSUE TITLE
`torch.xpu.amp.autocast` can cause NaN values in tensors

### ISSUE THREAD

**Issue by arekkulczycki** [intel/intel-extension-for-pytorch#539]
### Describe the bug

I'm using `stable-baselines3` for my training and running on Arc A770, therefore using `xpu` device.

```
import torch as th
import intel_extension_for_pytorch as ipex
from stable_baselines3 import PPO

model = PPO(my_policy, my_env)
with th.xpu.amp.autocast(enabled=True, dtype=th.bfloat16):
    model.learn()
```

The error shows up after many hours of training, that is, the training breaks complaining about `nan` values. My policy is using `torch.optim.AdamW` optimizer, which maybe is significant because the `nan` I was getting first shows up in the optimizer params during optimizer step. It seemed to only happen when loss was reaching high values like -40.

Once the error shows, then every time I rerun from the last checkpoint it shows again at the same moment. After removing the `autocast` line the training continues fine.

### Versions

PyTorch version: 2.1.0a0+cxx11.abi
PyTorch CXX11 ABI: Yes
IPEX version: 2.1.10+xpu
IPEX commit: a12f9f650
Build type: Release

OS: Manjaro Linux (x86_64)
GCC version: (GCC) 13.2.1 20230801
Clang version: 16.0.6
IGC version: 2024.0.0 (2024.0.0.20231017)
CMake version: version 3.28.1
Libc version: glibc-2.38

Python version: 3.11.6 (main, Nov 23 2023, 18:54:02) [GCC 13.2.1 20230801] (64-bit runtime)
Python platform: Linux-6.7.0-0-MANJARO-x86_64-with-glibc2.38
Is XPU available: True
DPCPP runtime version: 2024.0
MKL version: 2024.0
GPU models and configuration: 
[0] _DeviceProperties(name='Intel(R) Arc(TM) A770 Graphics', platform_name='Intel(R) Level-Zero', dev_type='gpu, support_fp64=0, total_memory=15473MB, max_compute_units=512, gpu_eu_count=512)
Intel OpenCL ICD version: N/A
Level Zero version: N/A

CPU:
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Address sizes:                      46 bits physical, 48 bits virtual
Byte Order:                         Little Endian
CPU(s):                             16
On-line CPU(s) list:                0-15
Vendor ID:                          GenuineIntel
Model name:                         12th Gen Intel(R) Core(TM) i5-12600K
CPU family:                         6
Model:                              151
Thread(s) per core:                 2
Core(s) per socket:                 10
Socket(s):                          1
Stepping:                           2
CPU(s) scaling MHz:                 88%
CPU max MHz:                        4900.0000
CPU min MHz:                        800.0000
BogoMIPS:                           7375.00
Flags:                              fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault cat_l2 cdp_l2 ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid rdt_a rdseed adx smap clflushopt clwb intel_pt sha_ni xsaveopt xsavec xgetbv1 xsaves split_lock_detect user_shstk avx_vnni dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp hwp_pkg_req hfi vnmi umip pku ospke waitpkg gfni vaes vpclmulqdq tme rdpid movdiri movdir64b fsrm md_clear serialize pconfig arch_lbr ibt flush_l1d arch_capabilities
Virtualization:                     VT-x
L1d cache:                          416 KiB (10 instances)
L1i cache:                          448 KiB (10 instances)
L2 cache:                           9.5 MiB (7 instances)
L3 cache:                           20 MiB (1 instance)
NUMA node(s):                       1
NUMA node0 CPU(s):                  0-15
Vulnerability Gather data sampling: Not affected
Vulnerability Itlb multihit:        Not affected
Vulnerability L1tf:                 Not affected
Vulnerability Mds:                  Not affected
Vulnerability Meltdown:             Not affected
Vulnerability Mmio stale data:      Not affected
Vulnerability Retbleed:             Not affected
Vulnerability Spec rstack overflow: Not affected
Vulnerability Spec store bypass:    Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:           Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:           Mitigation; Enhanced / Automatic IBRS, IBPB conditional, RSB filling, PBRSB-eIBRS SW sequence
Vulnerability Srbds:                Not affected
Vulnerability Tsx async abort:      Not affected

Versions of relevant libraries:
[pip3] intel-extension-for-pytorch==2.1.10+xpu
[pip3] numpy==1.26.2
[pip3] torch==2.1.0a0+cxx11.abi
[pip3] torchaudio==2.1.0a0+cxx11.abi
[pip3] torchvision==0.16.0a0+cxx11.abi
[conda] N/A

**Comment by arekkulczycki** [comment]
Sorry, as I finished reporting the issue I found that after more training time the `nan` popped up in the tensors regardless of removing `autocast`... Therefore I conclude it most likely has nothing to do with ipex.
